src/services/cache.ts

```diff
@@ -0,0 +1,10 @@
+import cache from 'cache';
+import {User, UserFilters} from "../types";
+
+export async function getCachedUsers(filters: UserFilters): Promise<User[] | undefined> {
+    return cache.get(getUsersCacheKey(filters));
+}
+
+function getUsersCacheKey(filters: UserFilters): string {
+    return `users-${filters.location}`;
```

Comment:
This is also potential problem, since we never validate that filters.location exists and is a string.